src/indicators/ElementSizeIndicator.C

```diff
@@ -0,0 +1,44 @@
+// libMesh includes
+#include "libmesh/numeric_vector.h"
+
+// MOOSE includes
+#include "Assembly.h"
+#include "MooseVariable.h"
+#include "SystemBase.h"
+
+// Mastodon includes
+#include "ElementSizeIndicator.h"
+
+registerMooseObject("MastodonApp", ElementSizeIndicator);
+
+InputParameters
+ElementSizeIndicator::validParams()
+{
+  InputParameters params = Indicator::validParams();
+  params.addParam<MaterialPropertyName>(
+      "element_size",
+      "element_size",
+      "The value of the material that will control the mesh refinement.");
+  params.addClassDescription("Computes the element size based on the provided material value.");
+  return params;
+}
+
+ElementSizeIndicator::ElementSizeIndicator(const InputParameters & parameters)
+  : Indicator(parameters),
+    _element_size(getMaterialProperty<Real>("element_size")),
```

Comment:
Why input the material property name if you are directly getting the property with the name "element_size"? 

this line should be 
`_element_size(getMaterialProperty<Real>(getParam<MaterialPropertyName>("element_size"))),`